### PR TITLE
add findLast string method

### PR DESCRIPTION
**Pull request by briandowns**
### What's Changed:

Add lastIndexOf string method to find the last index of a given string in another string.

### Type of Change:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [x] Tests have been updated to reflect the changes done within this PR (if applicable).
- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by Jason2605** [comment]
Thanks for this! Would it be possible to maybe change the name to `findLast`? Just thinking to keep it consistent with the forward looking `.find()`. Thinking as well (potentially a later PR) if we also want the optional skip like we have with `.find()`

**Comment by briandowns** [comment]
Thanks for the review. Changes made. I can do the optional skip in another PR.

File: src/vm/datatypes/strings.c [hunk 1]
```diff
@@ -275,6 +275,49 @@ static Value findString(DictuVM *vm, int argCount, Value *args) {
     return NUMBER_VAL(position);
 }
 
+static Value findLastString(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 1) {
+        runtimeError(vm, "lastInxedOf() takes 1 argument (%d given)", argCount);
```
Comment (Jason2605) [inline src/vm/datatypes/strings.c:280]:
```suggestion
        runtimeError(vm, "findLast() takes 1 argument (%d given)", argCount);
```

File: docs/docs/strings.md [hunk 2]
```diff
@@ -199,6 +199,15 @@ Returns true if a string contains another string.
 
 To find the index of a given substring, use the `.find()` method. This method takes an optional second parameter which can be used to skip the first `n` number of appearances of the substring. This method returns `-1` if the substring could not be found. Otherwise, it returns the index of the string.
 
+```cs
```
Comment (Jason2605) [inline docs/docs/strings.md:202]: I think there's a little bit of a mix up with the docs here, the find examples have went under `.findLast` and the `findLast` have went under `.find`

File: docs/docs/strings.md [hunk 3]
```diff
@@ -199,6 +199,15 @@ Returns true if a string contains another string.
...
+"woolly woolly mammoth".lastIndexOf("woolly"); // 7
+"mammoth".lastIndexOf("woolly"); // -1
```
Comment (Jason2605) [inline docs/docs/strings.md:204]:
```suggestion
"woolly woolly mammoth".findLast("woolly"); // 7
"mammoth".findLast("woolly"); // -1
```

File: tests/strings/findLast.du [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/**
+ * lastIndexOf.du
```
Comment (Jason2605) [inline tests/strings/findLast.du:2]:
```suggestion
 * findLast.du
```

File: tests/strings/findLast.du [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * Testing the str.lastIndexOf() method
```
Comment (Jason2605) [inline tests/strings/findLast.du:4]:
```suggestion
 * Testing the str.findLast() method
```

File: tests/strings/findLast.du [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * .lastIndexOf() returns the last index of the given string
```
Comment (Jason2605) [inline tests/strings/findLast.du:6]:
```suggestion
 * .findLast() returns the last index of the given string
```

**Review by Jason2605 — COMMENTED** [review]
Couple of small comments!

**Comment by Jason2605** [comment]
Thanks for this!

**Comment by briandowns** [comment]
Thanks for that commit. Apparently my editor's find and replace isn't quite working as expected. I needed this method for a text wrap module I wrote over the weekend. It's super small in Dictu but a pain to do it in C. Do we want to entertain string methods in Dictu like we have with lists and dicts?

**Comment by Jason2605** [comment]
No worries! Yeah built in methods on strings are more than welcome additions

**Comment by briandowns** [comment]
Awesome. Thanks!

**Pull request merged**